Assets/Text/JavaScript/NodeScripts/VelocityDampener.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Dampens the velocity components of a BodyNode.
+ * 
+ * Script written by Aiden Ravenshea.
+ */
+class VelocityDampener {
+
+    constructor(node) {
+        this.node = node;
+        this.linearForce = 5; // Linear velocity dampening force
+        this.angularForce = 5; // Angular velocity dampening force
+        this.scene = this.node.findSceneNode();
+    }
+
+    update(delta) {
+        // Get dampening forces and dampen velocity components
+        const fv = Math.min(1, this.linearForce * this.scene.timeStep);
```

Comment:
At this point scene is not guaranteed to be found. It might be undefined.
Please add a
`if (this.scene) { ... }`

Reply:
The new file now has this addition and the other one. Thanks!